utils/build-script-impl

```diff
@@ -409,58 +443,28 @@ function set_build_options_for_host() {
     local host="$1"
 
     # Hosts which can be cross-compiled must specify:
-    # SWIFT_HOST_TRIPLE and llvm_target_arch (as well as usual HOST_VARIANT flags)
+    # SWIFT_HOST_TRIPLE and llvm_target_arch.
+    # Hosts which have differing platforn names from their
+    # SWIFT_HOST_VARIANT_* values should change them here as well.
+
+    verify_host_is_supported $host
+
+    local platform=${host%%-*}
+    local architecture=${host##*-}
+
+    SWIFT_HOST_VARIANT=$platform
```

Comment:
How is the assignment for the simulator preserved?  The environment in the triple is different from the host variant I believe.

Reply:
Ugh, host is not the host in terms of build, its a custom tuple.  *sigh*